### DISCUSSION TITLE
Searching the database

### DISCUSSION THREAD

**Discussion by RichardUUU** [FreshRSS/FreshRSS#3610] [General]
I'd like to query the database to find out how many times my feeds include specific words. For example how many times in the past week has the word 'doghouse' been mentioned?

Any ideas on how I can do this?

**Comment by Alkarex** [comment]
Try in the search field: `doghouse date:P1w`
Do you mean the corresponding SQL query?

**Comment by RichardUUU** [comment]
Thanks for your reply. Rather than "see" the instances of my search word, I want to know that the word "doghouse" occurred say, 137 times this week. And I want to track that word every week, along with a hundred other words that I want to track every week.

SQL queries seem appropriate. I'm running the Docker version. How would I query the database?

**Reply by Alkarex** [reply]
What database are you using?

**Reply by RichardUUU** [reply]
I'm using the internal SQLite. But I have also created an instance of FreshRSS that uses MariaDB. I can see how to get to that data.

BTW, when starting a new instance, the FreshRSS UI asks which database to connect to. It would be nice if I could get back to that screen to change Databases.

**Reply by Alkarex** [reply]
Changing database requires a migration. What we offer is via the FreshRSS command line https://github.com/FreshRSS/FreshRSS/tree/edge/cli

Export from any database to SQLite, and then import from SQLite to any new database.

**Comment by Alkarex** [comment]
@RichardUUU 

## Using MySQL / MariaDB

```sh
docker exec -it your_mariadb mysql freshrss_db --user freshrss_user --password
```

```sql
SELECT count(*) FROM alice_entry WHERE id >= 1616522382081180 AND CONCAT(title, UNCOMPRESS(content_bin)) LIKE '%doghouse%';
```

This searches in title and content. 1616522382081180 is oldest unix time in μs. `alice_entry` is the table name, which depends on the name of your FreshRSS user.

You can also pass the SQL query directly:

```sh
docker exec your_mariadb mysql freshrss_db --user freshrss_user --password -e \
 "SELECT count(*) FROM alice_entry WHERE id >= 1616522382081180 AND CONCAT(title, UNCOMPRESS(content_bin)) LIKE '%doghouse%';"
```

## Using SQLite

```sh
sqlite3 /var/lib/docker/volumes/freshrss-data/_data/users/alice/db.sqlite \
 "SELECT count(*) FROM entry WHERE id >= 1616522382081180 AND CONCAT(title, content) LIKE '%doghouse%';"
```

Where `/var/lib/docker/volumes/freshrss-data/` is the path of your Docker volume.
